internal/ingestion/component/vision_enhancement.go

```
@@ -0,0 +1,288 @@
+			}
+			messages := buildVisionMessages(prompt, img)
+			resp, ierr := visionChatInvoker(ctx, driver, modelName, messages, apiConfig)
+			if ierr != nil {
+				return
+			}
+			descriptions[slot] = extractVisionAnswer(resp)
+		}(slot, tg.idx)
+	}
+	wg.Wait()
+	if ctx.Err() != nil {
+		return dispatched, false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 183) but it returns nil

(nilerr)